File: 백준 문제 풀이 모음/python/class 3/1697.py
import sys
from collections import deque

n, k = map(int, sys.stdin.readline().split())

def bfs() :
    q = deque()
    q.append(n)

    while q :
        x = q.popleft()
        if x == k  :
            print(dist[x])
            break

        for i in (x-1, x+1, x* 2 ) :
            if 0 <= i <= max and dist[i] == 0 :
                dist[i] = dist[x] + 1
                q.append(i)


max = 100000
dist = [0] * (max + 1)

bfs()

File: 백준 문제 풀이 모음/kotlin/class 3/1697.kt
package class3

var arr = Array<Int>(0) { 0 }
fun main () = with(System.`in`.bufferedReader()) {
    val str = readLine().split(' ').toList()
    val n = str[0].toInt()
    val k = str[1].toInt()
    arr = Array<Int> (100001) { 0 }

    BFS(n, k)

}

fun BFS(n: Int, k : Int) {
    val deque = ArrayDeque<Int>()
    deque.add(n)

    while (deque.size != 0) {
        val node = deque.removeFirst()
        if (node == k) {
            println(arr[node])
            break
        }

        val list = listOf<Int>(node-1, node+1, node * 2)
        list.forEach {
            if ( (0 <= it && it < 100001) && arr[it] == 0 ) {
                arr[it] = arr[node] + 1
                deque.add(it)
            }
        }


    }



}
